tart2ms/tart2ms.py

```diff
@@ -420,13 +477,17 @@ def __twelveball(direction):
                     [name], dtype=object), chunks=1)
                 dask_time = da.asarray(np.asarray(
                     [epoch_s_i], dtype=object), chunks=1)
-                dataset = Dataset({
-                    "NUM_LINES": (("row",), dask_num_lines),
-                    "NAME": (("row",), dask_name),
-                    "TIME": (("row",), dask_time),
-                    "DIRECTION": (("row", "dir"), dask_direction),
-                })
-                src_table.append(dataset)
+                all_numlines.append(dask_num_lines)
+                all_name.append(dask_name)
+                all_time.append(dask_time)
+                all_direction.append(dask_direction)
+        dataset = Dataset({
+            "NUM_LINES": (("row",), da.concatenate(all_numlines, axis=0).rechunk(-1)),
+            "NAME": (("row",), da.concatenate(all_name, axis=0).rechunk(-1)),
+            "TIME": (("row",), da.concatenate(all_time, axis=0).rechunk(-1)),
+            "DIRECTION": (("row", "dir"), da.concatenate(all_direction, axis=0).rechunk(-1)),
+        })
+        src_table.append(dataset)
```

Comment:
Necessaty to optimize graph construction otherwise the majority of the runtime is spent in locking and unlocking the tables for each append